proof/invariant-abstract/X64/ArchInvariants_AI.thy

```diff
@@ -1734,8 +1753,51 @@ lemma global_refs_update [iff]:
 
 end
 
+context Arch_p_asid_table_update_eq begin
+
+lemma vs_lookup_update[iff]:
+  "vs_lookup (f s) = vs_lookup s"
+  by (auto elim!: vs_lookup_stateI simp: asid_table obj_at_def)
+
+lemma vspace_at_asid_update[iff]:
+  "vspace_at_asid asid pt (f s) = vspace_at_asid asid pt s"
+  by (simp add: vspace_at_asid_def)
+
+lemma vs_lookup_pages_update[iff]:
+  "vs_lookup_pages (f s) = vs_lookup_pages s"
+  by (auto elim!: vs_lookup_pages_stateI simp: asid_table obj_at_def)
+
+lemma valid_vs_lookup_update [iff]:
+  "valid_vs_lookup (f s) = valid_vs_lookup s"
+  by (simp add: valid_vs_lookup_def asid_table )
+
+lemma valid_table_caps_update [iff]:
+  "valid_table_caps (f s) = valid_table_caps s"
+  by (simp add: valid_table_caps_def asid_table pspace second_level_tables)
+
+lemma valid_vspace_objs_update [iff]:
+  "valid_vspace_objs (f s) = valid_vspace_objs s"
+  by (simp add: valid_vspace_objs_def pspace)
+
+lemma valid_arch_caps_update [iff]:
+  "valid_arch_caps (f s) = valid_arch_caps s"
+  by (simp add: valid_arch_caps_def asid_table)
+
+lemma valid_kernel_mappings_update [iff]:
+  "valid_kernel_mappings (f s) = valid_kernel_mappings s"
+  by (simp add: valid_kernel_mappings_def pspace second_level_tables)
+
+lemma valid_asid_map_update [iff]:
+  "valid_asid_map (f s) = valid_asid_map s"
+  by (simp add: valid_asid_map_def)
```

Comment:
Something feels off about seeing these here... are these properties that we needed on other arches, but somehow got away with not needing on x64 before? Are they moved from somewhere else? Duplicated? It's part of `Arch_asid_table_update_eq` here.

Reply:
The high-level summary is that these locale changes are me backporting improvements from AARCH64, made a bit complicated due to not doing a complete job of it and focusing on the parts that I need now.

In this case, it's the addition of a locale that is specifically for functions that do not touch the kheap and asid_table. Some of the properties were then moved from a more general locale (where all of arch_state is preserved), while a couple of new properties were copied from AARCH64. The main benefit for me is that this new locale can be instantiated for `x64_current_fpu_owner_update` later on, giving me some useful lemmas for free.